v2.2/query-behavior-troubleshooting.md

```diff
@@ -1,31 +1,94 @@
 
-- [Identify and cancel long-running queries](manage-long-running-queries.html).
+You can then use the [Statements Details](https://www.cockroachlabs.com/docs/stable/admin-ui-statements-page.html#statement-details-page) page to drill down to individual statements. You can also use [`EXPLAIN ANALYZE`](https://www.cockroachlabs.com/docs/stable/explain-analyze.html) statement, which executes a SQL query and returns a physical query plan with execution statistics. Query plans provide information around SQL execution, which can be used to troubleshoot slow queries by figuring out where time is being spent, how long a processor (i.e., a component that takes streams of input rows and processes them according to a specification) is not doing work, etc.
 
-- [Turn on SQL logging](#sql-logging).
+If you need help interpreting the output of the `EXPLAIN ANALYZE` statement, contact us.
 
-If you're still unable to determine why the query executes slowly, please [file an issue](file-an-issue.html).
+## Query is sometimes slow
+
+If the query performance is irregular:
+
+1.  Run [`SHOW TRACE`](https://www.cockroachlabs.com/docs/stable/show-trace.html) for the query twice: once when the query is performing as expected and once when the query is slow.
+
+2.  Contact us to analyze the outputs of the `SHOW TRACE` command.
+
+## Cancelling running queries
+
+See [Cancel query](https://www.cockroachlabs.com/docs/v2.1/manage-long-running-queries.html#cancel-long-running-queries)
+
+## Low throughput
+
+Throughput is affected by the Disk I/O, CPU usage, and Network latency. Use the Admin UI to check the following metrics:
+
+Disk I/O: [Disk IOPS in progress](https://www.cockroachlabs.com/docs/v2.1/admin-ui-hardware-dashboard.html#disk-iops-in-progress)
+
+CPU usage: [CPU percent](https://www.cockroachlabs.com/docs/v2.1/admin-ui-hardware-dashboard.html#cpu-percent)
+
+Network latency: [Service latency SQL 99th percentile](https://www.cockroachlabs.com/docs/v2.1/admin-ui-overview-dashboard.html#service-latency-sql-99th-percentile)
+
+## Single hot node: One node has much higher resource usage than other nodes
+
+To determine if you have a hot node in your cluster, [access the Admin UI](admin-ui-access-and-navigate.html#access-the-admin-ui), click **Metrics** on the left, and navigate to the following graphs. Hover over each of the following graphs to see the per-node values of the metrics. If one of the nodes has a higher value, you have a hot node in your cluster.
+
+-   Replication dashboard > Average queries per store graph.
+
+-   Overview Dashboard > Service Latency graph
+
+-   Hardware Dashboard > CPU percent graph
+
+-   SQL Dashboard > SQL Connections graph
+
+-   Hardware Dashboard > Disk IOPS in Progress graph
+
+**Solution:**
```

Comment:
I think that breaking each of these sections down into a similar structure would really help users. I really like the idea of having a solution section. Is this something we could make consistent across all of the sections?

Reply:
I really wanted to do that,  but found that all sections do not lend themselves well to the structure. Also wasn't sure if folks would like the problem > explanation > solution structure. Definitely planning on restructuring the content in a follow-up PR.